File: StudentSystem/Models/DersKayit.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StudentSystem.Model
{
    public class DersKayit
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int ID { get; set; }
        [ForeignKey("Dersler")]
        public int DersID { get; set; }
        [ForeignKey("Ogrenci")]
        public int Ogrenci_ID { get; set; }
        public DateTime CreatedDate { get; set; }

        public Dersler Dersler { get; set; }
        public Ogrenci Ogrenci { get; set; }
    }
}

File: StudentSystem/ClientApp/src/app/models/derskayit.ts
import { Ders } from "./ders";
import { Ogrenci } from "./ogrenci";

export class DersKayit {
  id: number;
  dersId: number;
  ogrenciId: number;
  createdDate: number;
  dersler: Ders;
  ogrenci: Ogrenci;
}
